h2o-algos/src/main/java/hex/tree/dt/mrtasks/CountBinsSamplesCountsMRTask.java

```diff
@@ -64,25 +71,24 @@ public void map(Chunk[] cs) {
                     for (int i = 0; i < _bins.length; i++) {
                         // find bin by category
                         if (_bins[i][0] == cs[_featureSplit].atd(row)) {
```

Comment:
for lines 71 to 78, can't you just use cs[_featureSplit].atd(row) directly to index into _bins as 

int classIndex = (int) cs[_featureSplit].atd(row);
_bins[classIndex][_countsOffset]++;
_bins[classIndex][_countsOffset+1+classIndex]++;

You can replace the for loop with the above three lines.

Reply:
No, it wouldn't work as some categories can be missing in the nodes. For example if categories are '1','2','3' and in some node we only have data like (1,3,3,1,3), so we don't have a bin for category '2'. It was implemented like this in the beginning, but I fixed it as is was not correct.